design/crd-configstate.md

```diff
@@ -0,0 +1,275 @@
+# ConfigurationState CRD
+
+As a cluster user, I want to verify that MetalLB CRs successfully applied to the API
+are also successfully processed by MetalLB components. This CR surfaces configuration
+errors currently hidden in logs, making them visible via standard kubectl commands.
+
+
+## Configuration Errors
+
+We define configuration errors as errors caused solely by user input that can only be
+resolved by correcting that input. This is in contrast to infrastructure errors (such as
+missing RBAC permissions to watch resources) which can be fixed by administrators through
+deployment changes or code bug fixes.
+
+**Note:** Webhooks are covering almost all configuration errors but we can not
+rely because in corner cases (many calls in parallel) could two conflicted CRs
+stored in the API.
+
+**Note:** Specifically do not cover the case where the applied frrk8sconfig from
+speaker gives error, because of other external config in the frrk8s daemon.
+
+## API
+
+One CRD definitions, Multiple CRs, Namespaced, (maybe to configState)
+
+```
+$kubectl get configurationstate
+NAME         TYPE        NODE   VALID   LASTERROR   AGE
+controller   Controller         true                5m
+speaker-n1   Speaker     n1     true                5m
+speaker-n2   Speaker     n2     true                5m
+...
+```
+
+```go
+// ConfigurationStateSpec defines the desired state of ConfigurationState.
+type ConfigurationStateSpec struct {
+	// Type identifies whether this is a controller or speaker instance.
+	// +kubebuilder:validation:Enum=Controller;Speaker
+	Type string `json:"type"`
+
+	// NodeName is set when Type is "Speaker" to identify which node this speaker is running on.
+	// +optional
+	NodeName string `json:"nodeName,omitempty"`
+}
+
+// ConfigurationStateStatus defines the observed state of ConfigurationState.
+type ConfigurationStateStatus struct {
+	// ValidConfig indicates whether the configuration is valid.
+	// True when all reconcilers report success, False otherwise.
+	// +optional
+	ValidConfig bool `json:"validConfig,omitempty"`
```

Comment:
I'd have this as a string, that for now will only be Valid/Invalid (to be more flexible)

Reply:
added a commit https://github.com/metallb/metallb/pull/2870/commits/ff30a0e123c1846d981e70481ffdec5850c10f0b if we like the result I squash,